### PR TITLE
Find/Replace overlay: New/Noteworthy Entry

### PR DESCRIPTION
**Pull request by Wittmaxi**
N&N for https://github.com/eclipse-platform/eclipse.platform.ui/pull/1192

### CODE REVIEW

**Comment by HannesWell** [comment]
Thanks for providing this.

> I am not sure yet how I can correctly render this page and I am not quite sure how to word the N&N. If you could give me a few hints and/or review my text, i'd be very thankful :)

Sure. :) In general you have already found the right place and after a coarse view it looks good already.

The Contributing Guide leads you to more detailed instructions and recommendations about the content and how to validate it:
https://github.com/eclipse-platform/www.eclipse.org-eclipse/blob/0ef3497d054caaed1d4e335192ce0c563088db76/CONTRIBUTING.md

You can try it out by simply opening the modified html file in your browser. It does not look exactly as deployed to the website (I think the theming is missing), but it should give you a good general impression.

Once this is submitted it will be published at https://eclipse.dev/eclipse/news/4.33/platform.php shortly later.

When you think this is ready I can have a detailed review.

File: news/4.33/platform.html [hunk 1]
```diff
@@ -66,6 +66,30 @@ <h2>Views, Dialogs and Toolbar </h2>
...
+				<li><img style="max-width: 32px;" src="images/whole_word.svg"/> Whole word search (Ctrl + shift + W)</li>
+				<li><img style="max-width: 32px;" src="images/regex.svg"/> Search for a regular expression (Ctrl + shift + P)</li>
+				<li><img style="max-width: 32px;" src="images/search_in_area.svg"/> Search in a selected area (Ctrl + shift + A)</li>
+			</ul>
+		</p>
+		<p>In case you prefer the old find/replace dialog, you can enable it in the preferences under <code>General -> Editors -> Text Editors</code>
+			<img style="max-width: 600px;" src="images/overlay_top_replace_open.png" alt="Find/Replace overlay"/>
+			You can also choose to display the overlay at the bottom of the page.
+		</p>
+		<p><img style="max-width: 600px;" src="images/overlay_bottom.png" alt="Find/Replace overlay"/></p>
```
Comment (fedejeanne) [inline news/4.33/platform.html:89]:
```suggestion
		<p><img src="images/overlay_bottom.png" alt=""/></p>
```

File: news/4.33/platform.html [hunk 2]
```diff
@@ -66,6 +66,30 @@ <h2>Views, Dialogs and Toolbar </h2>
...
+		<p><img style="max-width: 600px;" src="images/overlay_top_replace_closed.png" alt="Find/Replace overlay"/></p>
```
Comment (fedejeanne) [inline news/4.33/platform.html:90]:
```suggestion
		<p><img src="images/overlay_top_replace_closed.png" alt=""/></p>
```

File: news/4.33/platform.html [hunk 3]
```diff
@@ -66,6 +66,30 @@ <h2>Views, Dialogs and Toolbar </h2>
...
+			The supported search options are:
+			<ul>
+				<li><img style="max-width: 32px;" src="images/case_sensitive.svg"/> Case sensitive search (Ctrl + shift + C)</li>
+				<li><img style="max-width: 32px;" src="images/whole_word.svg"/> Whole word search (Ctrl + shift + W)</li>
+				<li><img style="max-width: 32px;" src="images/regex.svg"/> Search for a regular expression (Ctrl + shift + P)</li>
+				<li><img style="max-width: 32px;" src="images/search_in_area.svg"/> Search in a selected area (Ctrl + shift + A)</li>
+			</ul>
+		</p>
+		<p>In case you prefer the old find/replace dialog, you can enable it in the preferences under <code>General -> Editors -> Text Editors</code>
+			<img style="max-width: 600px;" src="images/overlay_top_replace_open.png" alt="Find/Replace overlay"/>
```
Comment (fedejeanne) [inline news/4.33/platform.html:86]:
```suggestion
			<img src="images/overlay_top_replace_open.png" alt=""/>
```

File: news/4.33/platform.html [hunk 4]
```diff
@@ -66,6 +66,30 @@ <h2>Views, Dialogs and Toolbar </h2>
...
+		<p><img style="max-width: 600px;" src="images/overlay_top_replace_open.png" alt="Find/Replace overlay"/></p>
+		<p>The find/replace dialog now appears as an overlay on top of the editor. 
+			By default, it performs a "search as you type". This overlay includes several search options that can
+			be toggled on or off either through keyboard shortcuts or by clicking the corresponding icons in the overlay.
+			The supported search options are:
+			<ul>
+				<li><img style="max-width: 32px;" src="images/case_sensitive.svg"/> Case sensitive search (Ctrl + shift + C)</li>
+				<li><img style="max-width: 32px;" src="images/whole_word.svg"/> Whole word search (Ctrl + shift + W)</li>
+				<li><img style="max-width: 32px;" src="images/regex.svg"/> Search for a regular expression (Ctrl + shift + P)</li>
+				<li><img style="max-width: 32px;" src="images/search_in_area.svg"/> Search in a selected area (Ctrl + shift + A)</li>
```
Comment (fedejeanne) [inline news/4.33/platform.html:82]:
```suggestion
				<li><img src="images/case_sensitive.svg"/> Case sensitive search (<code>Ctrl</code> + <code>shift</code> + <code>C</code> in Windows)</li>
				<li><img src="images/whole_word.svg"/> Whole word search (<code>Ctrl</code> + <code>shift </code>+ <code>W</code> in Windows)</li>
				<li><imgsrc="images/regex.svg"/> Search for a regular expression (<code>Ctrl</code> + <code>shift</code> + <code>P</code> in Windows)</li>
				<li><img src="images/search_in_area.svg"/> Search in a selected area (<code>Ctrl</code> + <code>shift</code> + <code>A</code> in Windows)</li>
```

File: news/4.33/platform.html [hunk 5]
```diff
@@ -66,6 +66,30 @@ <h2>Views, Dialogs and Toolbar </h2>
     <h2>Text Editors </h2>
     </td>
   </tr>
+  
+  <tr id="find-replace-overlay"> <!-- https://github.com/eclipse-platform/eclipse.platform.ui/pull/1192 -->
+  	<td class="title">Overlay for performing find/replace operations</td>
+  	<td class="content">
+		<p><img style="max-width: 600px;" src="images/overlay_top_replace_open.png" alt="Find/Replace overlay"/></p>
```
Comment (fedejeanne) [inline news/4.33/platform.html:73]:
```suggestion
		<p><img src="images/overlay_top_replace_open.png" alt=""/></p>
```

**Review by fedejeanne — CHANGES_REQUESTED** [review]
Thank you for contributing this!
I just made some corrections based on the [instructions](https://eclipse.dev/eclipse/news/instructions.html).

Could you please also:
* Rename the files to use "-" instead of "_"
* Remove the "noise" (parts from buttons and texts in the background) from the edges of the screenshots

?

File: news/4.33/platform.html (hunk 5 above)
Comment (Wittmaxi) [reply news/4.33/platform.html]: https://github.com/eclipse-platform/www.eclipse.org-eclipse/commit/1a45b4636439d41d6d0ac95249a239328a397ff8

File: news/4.33/platform.html (hunk 4 above)
Comment (Wittmaxi) [reply news/4.33/platform.html]: https://github.com/eclipse-platform/www.eclipse.org-eclipse/commit/1a45b4636439d41d6d0ac95249a239328a397ff8

File: news/4.33/platform.html (hunk 3 above)
Comment (Wittmaxi) [reply news/4.33/platform.html]: https://github.com/eclipse-platform/www.eclipse.org-eclipse/commit/1a45b4636439d41d6d0ac95249a239328a397ff8

File: news/4.33/platform.html (hunk 2 above)
Comment (Wittmaxi) [reply news/4.33/platform.html]: https://github.com/eclipse-platform/www.eclipse.org-eclipse/commit/1a45b4636439d41d6d0ac95249a239328a397ff8

File: news/4.33/platform.html (hunk 1 above)
Comment (Wittmaxi) [reply news/4.33/platform.html]: https://github.com/eclipse-platform/www.eclipse.org-eclipse/commit/1a45b4636439d41d6d0ac95249a239328a397ff8

**Review by fedejeanne — CHANGES_REQUESTED** [review]
image

Also, I would rather have the screenshots in the "default" theme as opposed to the "dark" theme in order to be consistent across the document (see screenshots from previous feature).

**Comment by Wittmaxi** [comment]
@fedejeanne @HeikoKlare I have updated the PR according to your suggestions. I have anticipated the merge of the Find/Replace History and already documented it in this PR

**Comment by Wittmaxi** [comment]
grafik

**Comment by fedejeanne** [comment]
> @fedejeanne @HeikoKlare I have updated the PR according to your suggestions. 

* You didn't provide an empty `alt` text for the images

Additionally
* The word `shift` needs to be properly capitalized (`Shift`)
* I would remove the screenshot showing the overlay at the bottom and simply mention the fact that you can have at the bottom **before** the previous screenshot since that one already shows the option right below "_Use find/replace overlay_" anyway.

> I have anticipated the merge of the Find/Replace History and already documented it in this PR
👍

**Comment by HeikoKlare** [comment]
To have this N&N in soon and make people aware of the new feature, I have improved the N&N proposal according to the review and recent changes to the look and feel.
In more detail:
- Updated all screenshots with the current look and feel and proper cropping at the borders of the according shells.
- Removed the screenshot showing an overlay at the bottom of an editor. The editor was not completely shown so the screenshot looked a bit misplaced.
- Added a screenshot showing the expanded replace bar for comparison with the collapsed version.
- Added alternative descriptions for images.
- Made the shortcut references OS independent by referring to both the Windows/Linux and MacOS versions.
- Slightly improved the text.

A preview can be seen here: https://htmlpreview.github.io/?https://github.com/Wittmaxi/www.eclipse.org-eclipse/blob/MW_NN_find_replace_overlay/news/4.33/platform.html

@fedejeanne could you please have another look?

**Review by fedejeanne — CHANGES_REQUESTED** [review]
Remove the file `.png`, it's not necessary.

The rest looks good.

**Comment by HeikoKlare** [comment]
Removed the obsolete file in df372a9.

**Pull request merged**